src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs

```diff
@@ -26,38 +27,56 @@ public OrasClient(string imageReference)
             _imageReference = imageReference;
         }
 
-        public async Task PullImageAsync(string outputFolder)
+        public async Task<OCIOperationResult> PullImageAsync(string outputFolder)
         {
             string command = $"pull  \"{_imageReference}\" -o \"{outputFolder}\"";
-            await OrasExecutionAsync(command, Directory.GetCurrentDirectory());
+            string output = await OrasExecutionAsync(command, Directory.GetCurrentDirectory());
+            string digest;
+            try
+            {
+                digest = _digestRegex.Matches(output)[0].Groups["hex"].ToString();
+            }
+            catch (Exception ex)
+            {
+                throw new OrasException(TemplateManagementErrorCode.OrasProcessFailed, "Return image digest failed.", ex);
+            }
+
+            if (!Directory.Exists(outputFolder))
+            {
+                throw new OrasException(TemplateManagementErrorCode.OrasProcessFailed, "Pull image failed or image is empty.");
+            }
+
+            File.WriteAllText(Path.Combine(outputFolder, Constants.ManifestFileName), LoadManifestContentFromCache(digest));
+            return new OCIOperationResult() { ClientResponse = output };
```

Comment:
caller need to parse the output?

Reply:
For now, caller console.write the output.

Reply:
I feel the caller need to know more details instead of a simple string